rclpy/rclpy/client.py

```diff
@@ -37,6 +37,9 @@ def run(self):
 
         guard_condition_ready_list = \
             _rclpy.rclpy_get_ready_entities('guard_condition', self.wait_set)
+
+        # destroying here to make sure we dont call shutdown before cleaning up
+        _rclpy.rclpy_destroy_entity('guard_condition', sigint_gc)
```

Comment:
Won't this invalidate the variable `sigint_gc_handle` which is used next line?

Reply:
the pointer / handle is stored in a python int and is compared to a list of already populated int so I don't think that's a problem. But maybe I'm overseeing this

Reply:
I see, while the pointer is invalid the integer value is still the same and can be used for the following comparison. :+1: